crates/sui-framework/tests/safe_tests.move

```diff
@@ -144,4 +147,35 @@ module sui::safe_tests {
 
         ts::end(scenario_val);
     }
+
+    #[test]
+    #[expected_failure(abort_code = safe::TRANSFER_CAPABILITY_REVOKED)]
+    /// Ensure owner cannot withdraw funds after revoking itself.
+    fun test_safe_withdraw_self_revoked() {
+        let owner = TEST_OWNER_ADDR;
+        let scenario_val = ts::begin(owner);
+        let scenario = &mut scenario_val;
+
+        let initial_funds = 1000u64;
+        create_safe(scenario, owner, initial_funds);
+
+        ts::next_tx(scenario, owner);
+        let cap = ts::take_from_sender<OwnerCapability<SUI>>(scenario);
+        let safe = ts::take_shared<Safe<SUI>>(scenario);
+        let transfer_capability = safe::create_transfer_capability(&mut safe, &cap, initial_funds, ctx(scenario));
+        // Function under test
+        safe::self_revoke_transfer_capability(&mut safe, &transfer_capability);
```

Comment:
Not really related to your test, but why does `self_revoke_transfer_capability` accept the capability by reference, rather than by value, to make it impossible at a type system level to try and withdraw using a revoked capability?

Reply:
I love that tests are raising questions about interface params though :P I'm not sure. From how this is used, could it be because `object::id(...)` expects a reference?

https://github.com/MystenLabs/sui/blob/2343a71b84edcf04e27384b7d5d9b06e9504e06e/crates/sui-framework/sources/safe.move#L160-L164

Reply:
That's probably not the issue here, because if it were, you could take the capability by value, and take a reference of it locally.  One possibility is that it might be to do with the flow that allows a safe owner to revoke a transfer capability by ID.  If that happens, the capability will still exist, but is not usable.  Perhaps for uniformity, self-revoke is handled the same way?

It does seem strange to me that there is no way to burn the capability though.